### PR TITLE
CI: unpin myst-nb version

### PR DESCRIPTION
**Pull request by j-bowhay**
#### Reference issue

follow up to #19504
#### What does this implement/fix?

In #19504 we pinned the version of `myst-nb` in the doc CI job, testing if we can remove this

### CODE REVIEW

**Comment by j-bowhay** [comment]
Happy to open an issue upstream but beyond that the sphinx stuff is normally a bit over my head!

**Comment by lucascolley** [comment]
Can you try with Sphinx 7.2.6 and the latest version of the theme then?

**Comment by j-bowhay** [comment]
Hmm slightly confused in that with the latest version of Sphinx and the theme (which also puts myst-nb 1.0.0, see below) things work but the original issue came from a fresh environment which would have had those versions

```
- markdown-it-py    2.2.0  pyhd8ed1ab_0  conda-forge                  
  + markdown-it-py    3.0.0  pyhd8ed1ab_0  conda-forge/noarch     Cached
  - myst-nb          0.17.2  pyhd8ed1ab_0  conda-forge                  
  + myst-nb           1.0.0  pyhd8ed1ab_0  conda-forge/noarch     Cached
  - myst-parser      0.18.1  pyhd8ed1ab_0  conda-forge                  
  + myst-parser       2.0.0  pyhd8ed1ab_0  conda-forge/noarch     Cached
  - sphinx            5.3.0  pyhd8ed1ab_0  conda-forge                  
  + sphinx            7.2.6  pyhd8ed1ab_0  conda-forge/noarch     Cached
```

**Comment by lucascolley** [comment]
Yeah, doesn't really make sense to me. Can you try to reproduce the environment from a fresh install?

Either way, can we try removing the myst-nb pin from CI now?

**Comment by j-bowhay** [comment]
@melissawm @lucascolley I couldn't reproduce my original issue so I will try removing the pin in CI

**Comment by j-bowhay** [comment]
@melissawm @lucascolley CI is green so looks like we can, if we want, remove the pin in CI

File: .circleci/config.yml [hunk 1]
```diff
@@ -76,7 +76,6 @@ jobs:
           command: |
             pip install numpy cython pybind11 pythran ninja meson
             pip install -r requirements/doc.txt
-            pip install "myst-nb<1.0.0"
```
Comment (lucascolley) [inline .circleci/config.yml:79]: So it gets installed transitively right?

Comment (j-bowhay) [reply .circleci/config.yml:79]: Yes (although I didn't realise this at first hence all the force pushes!)

https://github.com/scipy/scipy/blob/2c6915ad7361d814621596afb12a4dd78603f77c/requirements/doc.txt#L9

**Review by lucascolley — APPROVED** [review]
LGTM, thanks Jake!

**Pull request merged**